Buildings/Experimental/DHC/EnergyTransferStations/Cooling/Indirect.mo

```diff
@@ -0,0 +1,314 @@
+  parameter Modelica.Units.SI.PressureDifference dpConVal_nominal(
+    final min=0,
+    displayUnit="Pa")=6000
+    "Nominal pressure drop of fully open control valve"
+    annotation(Dialog(group="Nominal condition"));
+  // Heat exchanger
+  parameter Modelica.Units.SI.PressureDifference dp1_nominal(
+    final min=0,
+    final start=500,
+    displayUnit="Pa")
+    "Nominal pressure difference on primary side"
+    annotation (Dialog(group="Heat exchanger"));
+  parameter Modelica.Units.SI.PressureDifference dp2_nominal(
+    final min=0,
+    final start=500,
+    displayUnit="Pa")
+    "Nominal pressure difference on secondary side"
+    annotation (Dialog(group="Heat exchanger"));
+  parameter Boolean use_Q_flow_nominal=true
+    "Set to true to specify Q_flow_nominal and temeratures, or to false to specify effectiveness"
+    annotation (Dialog(group="Heat exchanger"));
+  parameter Modelica.Units.SI.HeatFlowRate Q_flow_nominal(
+    final max=0,
+    final start=-100000)
+    "Nominal heat transfer"
+    annotation (Dialog(group="Heat exchanger"));
+  parameter Modelica.Units.SI.Temperature T_a1_nominal(
+    final min=0+273.15,
+    final max=100+273.15,
+    final start=5+273.15,
+    displayUnit="K")
+    "Nominal temperature at port a1"
+    annotation (Dialog(group="Heat exchanger"));
+  parameter Modelica.Units.SI.Temperature T_a2_nominal(
+    final min=0+273.15,
+    final max=100+273.15,
+    final start=7+273.15,
+    displayUnit="K")
+    "Nominal temperature at port a2"
+    annotation (Dialog(group="Heat exchanger"));
+  parameter Modelica.Units.SI.Efficiency eta(
+    final min=0,
+    final max=1)=0.8
+    "Constant effectiveness"
+    annotation (Dialog(group="Heat exchanger"));
+   //Controller parameters
+  parameter Real k(
+    final min=0,
+    final unit="1")=1
+    "Gain of controller"
+    annotation (Dialog(group="PID controller"));
+  parameter Modelica.Units.SI.Time Ti(
+    final min=Modelica.Constants.small)=120
+    "Time constant of integrator block"
+    annotation (Dialog(group="PID controller",enable=controllerType == CDL.Types.SimpleController.PI or controllerType == CDL.Types.SimpleController.PID));
+  parameter Modelica.Units.SI.Time Td(
+    final min=0)=0.1
+    "Time constant of derivative block"
+    annotation (Dialog(group="PID controller",enable=controllerType == CDL.Types.SimpleController.PD or controllerType == CDL.Types.SimpleController.PID));
+  parameter Real yMax(
+    final start=1)=1
+    "Upper limit of output"
+    annotation (Dialog(group="PID controller"));
+  parameter Real yMin=0.01
+    "Lower limit of output"
+    annotation (Dialog(group="PID controller"));
+  parameter Real wp(
+    final min=0)=1
+    "Set-point weight for Proportional block (0..1)"
+    annotation (Dialog(group="PID controller"));
+  parameter Real wd(
+    final min=0)=0
+    "Set-point weight for Derivative block (0..1)"
+    annotation (Dialog(group="PID controller"));
```

Comment:
If we decide to propagate all the PID parameters, then please make sure the parameters are enabled/disabled based on the choose of the controller type. Please refer the class `Buildings.Controls.Continuous.LimPID` for how to disable the parameters, like the parameters `Ni`, `Nd`, `xi_start`, etc.
In fact, the most commonly propagated parameters are `controllerType `, `k`, `Ti`, `Td`, `yMin` and `yMax`. The `reverseActing` should not be propagated as anytime we use the PID controller, this parameter should be fixed.

Reply:
Thanks for the suggestion. I have added enable condition for `Ti`, `Td`, `Ni`, `Nd`, `xi_start` and `xd_start`.  Also, like example in `Buildings.Controls.Continuous.Examples.LimPID`, only `controllerType` , `k`, `Ti`, `Td`, `yMin` and `yMax` will be propagated.